src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,53 +1,100 @@
 package core.basesyntax;
 
+import java.util.Arrays;
+import java.util.NoSuchElementException;
+
 /**
  * <p>Реалізувати свій ArrayList який імплементує інтерфейс List. Дотриматися основних вимог щодо
  * реалізації ArrayList (default capacity, newCapacity...)</p>
  */
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private T[] array;
+    private int size = 0;
+
+    public ArrayList() {
+        array = (T[]) new Object[DEFAULT_CAPACITY];
+    }
+
+    private void newCapacity() {
+        if (size >= array.length) {
+            array = Arrays.copyOf(array, array.length * 3 / 2);
+        }
+    }
+
+    private void checkForArrayIndexOutOfBoundsException(int index) {
+        if (index < 0 || index >= size) {
+            throw new ArrayIndexOutOfBoundsException();
+        }
+    }
 
     @Override
     public void add(T value) {
-
+        newCapacity();
+        array[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        checkForArrayIndexOutOfBoundsException(index);
+        if (size >= 0) {
+            System.arraycopy(array, 0, array, 1, size);
```

Comment:
arraycopy with this parameters will push all array for 1 position forward. is that exactly what we need here?